### PR TITLE
Additional error handling added for Create Volume

### PR DESCRIPTION
**Pull request by Madhan-SWE**
Added additional error handling for CreateVolume method.
These errors will be verified by unit tests.

### CODE REVIEW

File: pkg/cloud/cloud.go [hunk 1]
```diff
@@ -49,6 +49,9 @@ var (
 
 	// ErrAlreadyExists is returned when a resource is already existent.
 	ErrAlreadyExists = errors.New("resource already exists")
+
+	// ErrIdempotent is returned when another request with same idempotent token is in-flight.
```
Comment (mkumatag) [inline pkg/cloud/cloud.go:53]: ErrIdempotent is not matching the variable name

File: pkg/cloud/cloud.go [hunk 2]
```diff
@@ -49,6 +49,9 @@ var (
...
+	ErrIdempotentParameterMismatch = errors.New("parameters on this idempotent request are inconsistent with parameters used in previous request(s)")
```
Comment (mkumatag) [inline pkg/cloud/cloud.go:54]: where is this error used? are we returning this code anywhere in the driver?

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: pkg/cloud/cloud.go (hunk 2 above)
Comment (Madhan-SWE) [reply pkg/cloud/cloud.go:54]: Error is used here: https://github.com/ppc64le-cloud/powervs-csi-driver/pull/39/files#diff-4e4f8ad2f057d794bd5842ed50e358451e6e1b3b20e5f2836289665bdfbf5625R124

Comment (Madhan-SWE) [reply pkg/cloud/cloud.go:54]: fixed

**Comment by Madhan-SWE** [comment]
`ErrIdempotentParameterMismatch` was added as it was mainly tested in unit testing.
Needs further exploration as we are not returning this error from our createDisk at all.

**Comment by mkumatag** [comment]
> `ErrIdempotentParameterMismatch` was added as it was mainly tested in unit testing.
> Needs further exploration as we are not returning this error from our createDisk at all.

yes, exactly. Unless we return this error somewhere it is of no use of comparing anywhere, lets explore this further return the error wherever required with valid checks.

**Comment by Madhan-SWE** [comment]
`ErrIdempotentParameterMismatch` is not not returned by power-go-client.
The only error the client returns is `CreateVolumeOperationFailed`.
Reference: https://github.com/IBM-Cloud/power-go-client/blob/857f37685cdc44f06edffb3039f4c56a0b7332be/clients/instance/ibm-pi-volume.go#L54

This error is not applicable for the PowerVS. 
Can we close this PR?

**Comment by mkumatag** [comment]
> `ErrIdempotentParameterMismatch` is not not returned by power-go-client. The only error the client returns is `CreateVolumeOperationFailed`. Reference: https://github.com/IBM-Cloud/power-go-client/blob/857f37685cdc44f06edffb3039f4c56a0b7332be/clients/instance/ibm-pi-volume.go#L54
> 
> This error is not applicable for the PowerVS. Can we close this PR?

make sense, lets close

/close

**Pull request closed without merging**